### ISSUE TITLE
filter lists not being updated / don't contain all rules

### ISSUE THREAD

**Issue by DetachHead** [AdguardTeam/AdguardForiOS#1974]
### Issue Details

- **AdGuard version:**
  - AdGuard/AdGuard Pro for iOS v4.3.3(947)
- **Device model and storage size:**
  - iphone 13 128gb
- **Operating system and version:**
  - ios 15.1.1
- **Browser or App:**
  - safari

### Expected Behavior

same number of rules when re-adding the list as a custom filter list

### Actual Behavior

the easylist that came with the app only has 12219 rules but when i re-add easylist as a custom filter list there are 35730 rules

### Screenshots

<details><summary>the easylist that came with the app:</summary>
it appears to have never updated since i downloaded the app months ago (22 Dec 2021)

![image](https://user-images.githubusercontent.com/57028336/163064339-4a0697be-f287-4975-8e69-f639661091c3.png)

after disabling and re-enabling it it seems to have updated but it only gained a few more rules:

![image](https://user-images.githubusercontent.com/57028336/163064288-4358ab1f-4a07-4b64-80a9-e6b3d3c10330.png)

</details>
<details><summary>after re-adding easylist as a custom filter list from the same url:</summary>

![image](https://user-images.githubusercontent.com/57028336/163064477-c63ff2e8-3913-42e3-b8f1-13409920976d.png)
![image](https://user-images.githubusercontent.com/57028336/163064511-5683ff8b-ad1e-4d75-9151-389e9e0be42e.png)

</details>

### Additional Information
my filter lists don't seem to be getting updated, and when they do, it's only adding a fraction of the rules. as a result, i've started seeing ads again on many different sites

**Comment by ethan-xd** [comment]
“+1 I also experiencing same issue ….”

**Comment by Vitalik93** [comment]
Интересно услышать ответ разработчиков. Сам заметил что на IOS стало паршиво блокироваться реклама. Думал из за того, что пока стоит Adguard для iOS 14 без нового функционала для 15 версии, теперь уже сомневаюсь. Если такое такое количество правил отсутсвует, понятно в чем дело.

**Comment by zzebrum** [comment]
Hi! We use optimized filters by default. What it means and how we do so: https://kb.adguard.com/en/general/filter-rules-statistics#why-do-we-need-statistics
And when you add filter manually it is not optimized. That's why there is a difference. 

But there is a bug with filters update date.

**Comment by zzebrum** [comment]
what we can do is change this optimization algorithm a little.

**Comment by Vitalik93** [comment]
> Hi! We use optimized filters by default. What it means and how we do so: https://kb.adguard.com/en/general/filter-rules-statistics#why-do-we-need-statistics And when you add filter manually it is not optimized. That's why there is a difference.
> 
> But there is a bug with filters update date.

А если оптимизированные фильтры выключены, почему бы тогда все правила не применять?

**Comment by Alex-302** [comment]
Различие в количестве правил потому, что в приложении используется оптимизированный фильтр, а пользователь из первого сообщения видать импортировал полный. И подсчёт количества правил отличается от других Адгардов.

**Comment by maxikuzmin** [comment]
@DetachHead issue is still relevant?

**Comment by DetachHead** [comment]
@maxikuzmin yes i have to keep manually opening adguard and tapping the update button otherwise the lists never get automatically updated

**Comment by maxikuzmin** [comment]
@DetachHead about filter updates, the problem is known, it does not repeat so often among users, and we are already solving this problem.
Filters are usually refreshed within 24 to 72 hours. We recommend that you refresh the filters manually before this problem is fixed

**Comment by DetachHead** [comment]
@maxikuzmin thanks! is there already an open issue tracking this problem?

**Comment by maxikuzmin** [comment]
@DetachHead  this issue will be the main
